src/components/permissions/routes.ts

```diff
@@ -99,20 +99,46 @@ const routeUserPermissions = async (request: any, reply: any) => {
   const isSystemOrg = await databaseConnect.isInternalOrg(Number(orgId))
 
   const templatePermissionRows = await databaseConnect.getTemplatePermissions(isSystemOrg)
-  const userExistingPermissions = username
-    ? await databaseConnect.getUserTemplatePermissions(username, orgId)
-    : await databaseConnect.getOrgTemplatePermissions(isSystemOrg)
-
-  const grantedPermissions = userExistingPermissions.reduce(
-    (grantedPermissions, { permissionName }) => {
-      if (!grantedPermissions.includes(permissionName)) grantedPermissions.push(permissionName)
-      return grantedPermissions
-    },
-    []
-  )
+
+  let grantedPermissions: string[] = []
+  let availablePermissions: string[] = []
+
+  if (username) {
+    // Get permissions for organisation and which have been granted to user
+    const userExistingPermissions = await databaseConnect.getUserTemplatePermissions(
+      username,
+      orgId
+    )
+    console.log('userExistingPermissions', userExistingPermissions)
+
+    grantedPermissions = userExistingPermissions.reduce(
+      (grantedPermissions, { permissionName }) => {
+        if (!grantedPermissions.includes(permissionName)) grantedPermissions.push(permissionName)
+        return grantedPermissions
+      },
+      []
+    )
```

Comment:
```suggestion
 grantedPermissions = Array.from(new Set(userExistingPermissions.map((p) => p.permissionName))).sort()
```
Personally I find this easier to understand (and using Sets to generate lists of unique items is more efficient than searching through the same array over and over again)
Note also the additional `.sort()` these lists weren't coming back in alphabetical order

Reply:
Cool. That's a good idea : )